src/borg/platform/base.py

```diff
@@ -283,3 +291,22 @@ def process_alive(host, pid, thread):
 def local_pid_alive(pid):
     """Return whether *pid* is alive."""
     raise NotImplementedError
+
+
+def get_free_space(path):
+    if not is_win32:
+        try:
+            st_vfs = os.statvfs(path)
+        except OSError as os_error:
+            logger.warning('Failed to check free space before committing: ' + str(os_error))
+            return
+        except AttributeError:
+            # TODO move the call to statvfs to platform
+            logger.warning('Failed to check free space before committing: no statvfs method available')
+            return
+        # f_bavail: even as root - don't touch the Federal Block Reserve!
+        free_space = st_vfs.f_bavail * st_vfs.f_frsize
+    else:
+        import shutil
+        free_space = shutil.disk_usage(path).free
```

Comment:
hmm, did you check what the difference is between the above linux/posix code and the shutil based code? could we just use shutil for both?

Reply:
Just checked. `shutil.disk_usage` does exactly the same (minus the error checks) as the code which is used by borg. See https://github.com/python/cpython/blob/master/Lib/shutil.py#L1215.
Should I replace `get_disk_usage(path)` with `shutil.disk_usage(path).free`?

Reply:
yes, but make sure the code does not crash.